sip-quickstart/assets/admin/shared.private.js

```diff
@@ -0,0 +1,130 @@
+const crypto = require("crypto");
+const path = require("path");
+
+// Change the salt to invalidate tokens
+const SALT = "salty";
+
+// Creates a token for client side usage
+function createToken(context, password) {
+  const tokenString = `${context.ACCOUNT_SID}:${password}:${SALT}`;
+  // Similar to TwilioClient
+  return crypto
+    .createHmac("sha1", context.AUTH_TOKEN)
+    .update(Buffer.from(tokenString, "utf-8"))
+    .digest("base64");
+}
+
+function isAllowed(context, token) {
+  // Create the token with the environment password
+  const masterToken = createToken(context, process.env.ADMIN_PASSWORD);
+  return masterToken === token;
+}
+
+// Shortcuts by calling the callback with an error
+function checkAuthorization(context, event, callback) {
+  if (!isAllowed(context, event.token)) {
+    const response = new Twilio.Response();
+    response.setStatusCode(403);
+    response.setBody("Not authorized");
+    callback(null, response);
+    return false;
+  }
+  return true;
+}
+
+async function getCurrentEnvironment(context) {
+  if (context.DOMAIN_NAME && context.DOMAIN_NAME.startsWith("localhost")) {
+    return;
+  }
+  const client = context.getTwilioClient();
+  const services = await client.serverless.services.list();
+  for (let service of services) {
+    const environments = await client.serverless
+      .services(service.sid)
+      .environments.list();
+    const environment = environments.find(
+      env => env.domainName === context.DOMAIN_NAME
+    );
+    if (environment) {
+      // Exit the function
+      return environment;
+    }
+  }
+}
+
+async function getEnvironmentVariables(context, environment) {
+  const client = context.getTwilioClient();
+  return await client.serverless
+    .services(environment.serviceSid)
+    .environments(environment.sid)
+    .variables.list();
+}
+
+async function getEnvironmentVariable(context, environment, key) {
+  const client = context.getTwilioClient();
```

Comment:
Minor, but this function doesn't use this client.

Reply:
Thanks!

Reply:
Fixed!